corehq/ex-submodules/casexml/apps/phone/cleanliness.py

```diff
@@ -135,20 +136,43 @@ def get_cleanliness_flag_from_scratch(domain, owner_id):
         cases_to_check = cases_to_check - closed_owned_case_ids
         if cases_to_check:
             # it wasn't in any of the open or closed IDs - it must be dirty
-            reverse_index_ids = set(get_reverse_indexed_case_ids(domain, list(cases_to_check)))
```

Comment:
this function is pretty dense. not sure how to avoid that or whether we should just accept that this is complicated